tez-common/src/main/java/org/apache/tez/common/web/ProfileOutputServlet.java

```diff
@@ -0,0 +1,52 @@
+package org.apache.tez.common.web;
+
+import org.apache.hadoop.yarn.webapp.MimeType;
+import org.eclipse.jetty.servlet.DefaultServlet;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class ProfileOutputServlet extends DefaultServlet {
+    public static final String fileQueryParam = "file";
+    public ProfileOutputServlet() {
+
+    }
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String queriedFile = request.getParameter(fileQueryParam);
+        if (queriedFile == null) {
+            response.setContentType(MimeType.TEXT);
```

Comment:
you can refactor these lines to a single method (message string as parameter) and reuse it below